### PR TITLE
Stability, dev env and security improvements

### PR DESCRIPTION
**Pull request by smorimoto**
Add or change some things to make the project easier to manage and maintain.

- Use pnpm as the package manager
  - It's more stable, efficient, and a bit more strict than yarn v1
  - https://pnpm.io/npmrc#node-linker
- Let CI ensure that the contents of repository dist and “actual” dist match
  - It's important to ensure that PR from external contributors doesn't inject vulnerabilities
- Use Dependabot to keep GHA and npm dependencies up-to-date
- Use Biome to ensure adherence to code styles and best practices
- Update all dependencies

### CODE REVIEW

File: .github/dependabot.yml
Comment (smorimoto) [inline .github/dependabot.yml:1]: Keep all dependencies up-to-date

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+jobs:
+  hygiene:
+    name: Hygiene
+
+    strategy:
+      fail-fast: false
+      matrix:
+        action:
+          - setup-flyctl
```
Comment (smorimoto) [inline .github/workflows/workflow.yml:17]: If you add more sub-actions in the future, just add them here.

File: setup-flyctl/biome.json
Comment (smorimoto) [inline setup-flyctl/biome.json:1]: Not only does Biome perform a wider range of checks than Prettier, it's also a lot faster.

File: setup-flyctl/src/main.ts [hunk 2]
```diff
@@ -1,50 +1,62 @@
...
+import * as core from "@actions/core";
+import * as http from "@actions/http-client";
+import * as toolCache from "@actions/tool-cache";
+import { ARCHITECTURE, PLATFORM, VERSION } from "./constants.js";
 
-const client = new http.HttpClient('setup-flyctl')
+const client = new http.HttpClient("setup-flyctl", [], {
+  allowRetries: true,
+  maxRetries: 3,
+});
```
Comment (smorimoto) [inline setup-flyctl/src/main.ts:9]: Retries may be needed.

**Comment by jsierles** [comment]
Thanks for the PR! There are quite a lot of unrelated changes in here. Can you pull those into separate PRs, keeping this one focused on the architecture support?

File: setup-flyctl/src/main.ts [hunk 3]
```diff
@@ -1,50 +1,62 @@
...
+    core.addPath(toolPath);
   } else {
-    core.info(`Downloading flyctl ${resolvedVersion} from ${url}...`)
-    await installFlyctl(url, resolvedVersion)
+    core.info(`Downloading flyctl ${resolvedVersion} from ${url}...`);
+    const downloadPath = await toolCache.downloadTool(url);
+    const extractedPath =
+      PLATFORM === "Windows"
+        ? await toolCache.extractZip(downloadPath)
+        : await toolCache.extractTar(downloadPath);
```
Comment (smorimoto) [inline setup-flyctl/src/main.ts:38]: On Windows, it's zip not tar.

File: setup-flyctl/tsconfig.json [hunk 4]
```diff
@@ -1,15 +1,13 @@
 {
-  "extends": "@tsconfig/node20/tsconfig.json",
+  "extends": [
+    "@tsconfig/strictest/tsconfig.json",
+    "@tsconfig/node20/tsconfig.json"
+  ],
```
Comment (smorimoto) [inline setup-flyctl/tsconfig.json:5]: Just make it stricter.

File: setup-flyctl/package.json [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
-    "typescript": "^5.2.2"
-  }
+    "@biomejs/biome": "1.8.3",
+    "@tsconfig/node20": "20.1.4",
+    "@tsconfig/strictest": "2.0.5",
+    "@types/node": "20.14.12",
+    "@vercel/ncc": "0.38.1",
+    "typescript": "5.5.4"
+  },
+  "packageManager": "yarn@4.3.1"
```
Comment (smorimoto) [inline setup-flyctl/package.json:22]: corepack and yarn v3 are even faster and more stable.

**Comment by smorimoto** [comment]
@jsierles Ooh, I just wrote some comments to make the review easier. Do you still want to split? (That's fine 🙃)

**Comment by jsierles** [comment]
While I think some of these changes are good, it would be best to keep them in separate PRs.

**Comment by smorimoto** [comment]
@jsierles Done! #64

**Comment by smorimoto** [comment]
CI results: https://github.com/smorimoto/flyctl-actions/actions/runs/10113801238

**Comment by smorimoto** [comment]
@jsierles Ready for review!

**Comment by jsierles** [comment]
Can you update the PR description to summarize the changes here?

**Comment by smorimoto** [comment]
@jsierles Done!

**Pull request merged**